src/schema/v2/RenameFields.ts

```diff
@@ -121,7 +130,8 @@ export class RenameFields implements Transform {
             }
 
             const type = getTypeWithSelectableFields(typeInfo)
-            const oldName = this.changedFields[fieldKey(type, node.name.value)]
+            const oldName =
+              type && this.changedFields[fieldKey(type, node.name.value)]
```

Comment:
Why would there ever not be a type?

Reply:
yes, i was little surprised by that too, but in case of 👇 query, `__id`'s `type` is `undefined`
```graphql
{
  artwork(id: "retna-original-limited-edition-skateboard-skate-deck-blue-with-coa-signed-by-retna") {
    id
    gravityID
    shows{
      id
      gravityID
      name
    }
  }
}
```
```console
// debug console
node.name.value
"__id"
type
undefined
````

Reply:
actually with little more digging, looks like any field under `shows` has would have `undefined` type:
![image](https://user-images.githubusercontent.com/1230819/60512671-30923000-9ca3-11e9-8493-f2ec25d988da.png)

Reply:
Man, this `TypeInfo` API and its state during AST traversal. I should spend some time really understanding its implementation. Anyways, if tests pass and queries work–great job! 🎉